src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -321,5 +469,30 @@ public RunningFunctionInfo(string name, Guid invocationId, DateTime startTime, b
                 public bool Success { get; set; }
             }
         }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                {
+                    // flush any outstanding events
+                    TimerFlush(state: null);
+
+                    if (_metricsFlushTimer != null)
```

Comment:
I tried that initially, but that syntax causes our fxcop/style cop rules to say we're NOT calling Dispose. Any ideas?